superset/datasets/models.py

```diff
@@ -66,6 +67,15 @@ class Dataset(Model, AuditMixinNullable, ExtraJSONMixin, ImportExportMixin):
     # model has been deleted this column can be removed.
     sqlatable_id = sa.Column(sa.Integer, nullable=True, unique=True)
 
+    # default values set for filtering + querying if self.expression doesn't explicitly state
+    # which db or schema to read from
+    default_schema = sa.Column(sa.Text)
+    database_id = sa.Column(sa.Integer, sa.ForeignKey("dbs.id"), nullable=False)
+    default_database: Database = relationship(
+        "Database",
+        foreign_keys=[database_id],
```

Comment:
wouldn't we pull this information from the tables? cc @betodealmeida

Reply:
initially we would on create, but the main issue is being able to filter/query via the `ModelRestApi`

the class currently doesn't allow us to join across models, which is making it difficult to for me to get the filtering to work properly for these columns. So the idea was to add default column for both schema and database to allow us to be able to power the CRUD filtering and we'd allow the user to update this field through settings.

Reply:
@eschutho I talked with Hugh about this, we think we need the concept of a 'default database'. Right now there's an implicit assumption that all tables belong to the same database, but having a default database would allow us to relax that restriction in the future.

It's similar with schema, tables can belong to different schemas today, but there's still a concept of a default schema which is needed if the query references only the table name.